### ISSUE TITLE
Sync from UI for Multi source application stopped working on v2.11.0

### ISSUE THREAD

**Issue by stijnbrouwers** [argoproj/argo-cd#18120]
**Describe the bug**
Since the update to ArgoCD v.2.11.0 my multisource application is not syncing anymore. Reverting back to 2.10.9 makes it work again. The error message is:
```
Unable to sync: error resolving repo revision: rpc error: code = Unknown desc = repository not found
```
Single source application using the public repo works and single source application to the private git repo also still works. The issue only occurs for multisource applications combining the 2.

**To Reproduce**
Create an application that uses a public helm chart and retrieves values files from a private github repository:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: argocd
  namespace: argocd
spec:
  destination:
    namespace: argocd
    server: https://kubernetes.default.svc
  project: default
  sources:
  - chart: argo-cd
    helm:
      passCredentials: true
      valueFiles:
      - $values/values.yaml
    repoURL: https://argoproj.github.io/argo-helm
    targetRevision: 6.8.0
  - ref: values
    repoURL: https://github.com/org/some-private-repo
    targetRevision: develop
  syncPolicy:
    automated: {}
```

**Expected behavior**
I would expect the application to be able to sync

**Screenshots**

**Version**

```shell
argocd: v2.11.0+d3f33c0
  BuildDate: 2024-05-07T16:01:41Z
  GitCommit: d3f33c00197e7f1d16f2a73ce1aeced464b07175
  GitTreeState: clean
  GoVersion: go1.21.9
  Compiler: gc
  Platform: linux/amd64
```

**Logs**
```
time="2024-05-08T08:16:41Z" level=warning msg="finished unary call with code FailedPrecondition" error="rpc error: code = FailedPrecondition desc = error resolving repo revision: rpc error: code = Unknown desc = repository not found" │
│  grpc.code=FailedPrecondition grpc.method=Sync grpc.service=application.ApplicationService grpc.start_time="2024-05-08T08:16:41Z" grpc.time_ms=214.932 span.kind=server system=grpc                                                       │
│                                                                                                                                                     
```

**Comment by todaywasawesome** [comment]
@pasha-codefresh @ishitasequeira possible 2.11 regression in multi-source apps.

**Comment by solidnerd** [comment]
I don't know if it helps since I can't rember right now if it was also on public repositories. We use mostly everywhere private repositories and we had the issue on a private repository that was an oci helm-chart.

**Comment by avo-sepp** [comment]
I am seeing it on Applications that are multi-source mix of Private OCI Repository which stores the Helm Chart (Azure Container Registry) and Private Git Repository w/ Helm Values File. I am seeing it whenever a manual sync is attempted. However, I have seen a manual sync work after modifying the Git repository (new SHA hash available).

**Comment by avo-sepp** [comment]
Confirming I had to roll-back to 2.10.9 to get our teams working again. Sync and Auto-sync both failing.

**Comment by mnorrsken** [comment]
Happens to me too but saw it only on one application so far. I can't sync my gitlab application it uses a public helm chart and then a private github repository to provide the values file. Works again when rolling back to 2.10.9.
```yaml
project: default
destination:
  namespace: gitlab
  name: in-cluster
syncPolicy:
  syncOptions:
    - CreateNamespace=true
    - ApplyOutOfSyncOnly=true
sources:
  - repoURL: 'https://charts.gitlab.io'
    targetRevision: 7.*
    helm:
      valueFiles:
        - $values/gitlab/values.yaml
    chart: gitlab
  - repoURL: 'https://github.com/xxxxx/yyyyy.git' # this is a private repo
    path: gitlab
    targetRevision: main
    ref: values
```

**Comment by adecchi-2inno** [comment]
The fix will be available in the next release ?

**Comment by stefanandres** [comment]
The fix is in the released 2.11.1 release.
